pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,241 @@
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+)
+
+func newPackagePublishCmd() *cobra.Command {
+
+	var err error
+	var path string
+	var tag string
+
+	cmd := &cobra.Command{
+		Use:   "publish-sdk <package> <language>",
+		Args:  cobra.ExactArgs(2),
+		Short: "Publish a package SDK to supported package registries.",
+		Long:  "Publish a package SDK to supported package registries.",
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// Validate the language
+
+			lang := args[1]
+
+			switch lang {
+			case "all":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "nodejs":
+				fmt.Println("Nodejs is supported, publishing")
+			case "python":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "java":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			case "dotnet":
+				fmt.Println("This command is still under construction. " +
+					"Please wait for a future release.")
+			default:
+				return fmt.Errorf(
+					"language %s is not supported. Must be one of: \n"+
+						"- all \n"+
+						"- nodejs \n"+
+						"- python \n"+
+						"- java \n"+
+						"- dotnet ", lang,
+				)
+			}
+
+			return nil
+		},
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			pkg := args[0]
+			lang := args[1]
+
+			err = publish(pkg, lang, path, tag)
+			if err != nil {
+				return fmt.Errorf("error publishing: %s", err.Error())
+			}
+			return nil
+		}),
+	}
+	cmd.PersistentFlags().StringVar(&path, "path", "",
+		`The path to the root of your package.
+	Example: ./sdk/nodejs
+	`)
+	cmd.PersistentFlags().StringVar(&tag, "tag", "",
+		`The tag for your NPM package, such as "latest""`)
+	return cmd
+}
+
+func publish(pkg, lang, path, tag string) error {
+
+	fmt.Printf("You are publishing package %s to the %s registry \n", pkg, lang)
+	switch lang {
+	case "nodejs":
+		err := publishToNPM(pkg, path, tag)
+		if err != nil {
+			return err
+		}
+	case "python":
+		publishToPyPI(pkg)
+	case "java":
+		publishToMaven(pkg)
+	case "dotnet":
+		publishToNuget(pkg)
+	default:
+		// Publish to all registries
+		err := publishToNPM(pkg, path, tag)
+		if err != nil {
+			return err
+		}
+		publishToPyPI(pkg)
+		publishToMaven(pkg)
+		publishToNuget(pkg)
+	}
+
+	return nil
+}
+
+func publishToNPM(pkg, path, tag string) error {
+	// this is likely where we need to go get the sdk info
+	// verify path
+	_, err := os.Stat(path)
+	if errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("could not find path to package directory: %s", path)
+	}
+
+	// Verify  npm exists set up: npm, user login
+	ex, err := executable.FindExecutable("npm")
+	if err != nil {
+		return fmt.Errorf("could not find npm: %w", err)
+	}
+
+	// TODO: Verify auth for npm
+
+	whoamiCmd := exec.Command(ex, "whoami")
```

Comment:
It's probably best to point the stderr of this command to stderr so that the npm error message also gets printed to the user.

```go
whoamiCmd.Stderr = os.Stderr
// or, if packagePublishCmd has a stderr field,
//
// whoamiCmd.Stderr = cmd.Stderr
```

Reply:
ah, I did not realize you could do this. 👍